src/Data/Int.purs

```diff
@@ -77,6 +79,21 @@ foreign import toNumber :: Int -> Number
 fromString :: String -> Maybe Int
 fromString = fromStringAs (Radix 10)
 
+-- | A type for describing whether an integer is even or odd.
+data Parity = Even | Odd
+
+derive instance eqParity :: Eq Parity
+derive instance ordParity :: Ord Parity
```

Comment:
I don't think `&&` and `||` on booleans form a ring, as the operation you choose for addition has to form a group on booleans, and I don't think either of `&&` or `||` does; there's no inverse for false under `&&`, and no inverse for true under `||`.